### PR TITLE
Add tests for wstring conversion routines

### PR DESCRIPTION
**Pull request by cottsay**
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10815/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6193/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8806/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10702/)

### CODE REVIEW

File: rosidl_typesupport_fastrtps_cpp/test/test_wstring_conversion_mem.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    osrf_testing_tools_cpp::memory_tools::uninitialize();
+  });
+
+  std::wstring wstring(L"¡Hola, Mundo!");
+  std::u16string u16string(1, 1);
+
+  // Force malloc/realloc to fail
+  auto on_unexpected_operation =
+    []() {
+      throw std::exception();
```
Comment (brawner) [inline rosidl_typesupport_fastrtps_cpp/test/test_wstring_conversion_mem.cpp:39]: If this throws the developer will get a backtrace so they can know whether it was line 41 or 42 yah?

File: rosidl_typesupport_fastrtps_c/test/test_wstring_conversion.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#include <string>
+
+#include "rosidl_typesupport_fastrtps_c/wstring_conversion.hpp"
+
+using rosidl_typesupport_fastrtps_c::u16string_to_wstring;
+using rosidl_typesupport_fastrtps_c::wstring_to_u16string;
+
+TEST(test_wstring_conversion, wstring_round_trip)
+{
+  std::wstring wstring(L"¡Hola, Mundo!");
```
Comment (brawner) [inline rosidl_typesupport_fastrtps_c/test/test_wstring_conversion.cpp:28]: Add tests for empty string, and default constructed string. Are there any other potential failure cases that would need to be checked?

File: rosidl_typesupport_fastrtps_cpp/test/test_wstring_conversion_mem.cpp (hunk 1 above)
Comment (cottsay) [reply rosidl_typesupport_fastrtps_cpp/test/test_wstring_conversion_mem.cpp:39]: This isn't straightforward, but `osrf_testing_tools_cpp::memory_tools` will catch this exception, print an error message, and make the memory operation return `nullptr`. We should never see the backtrace.

File: rosidl_typesupport_fastrtps_c/test/test_wstring_conversion.cpp (hunk 2 above)
Comment (cottsay) [reply rosidl_typesupport_fastrtps_c/test/test_wstring_conversion.cpp]: I can think of no more than what you mentioned. Added in 38986f0.

**Comment by cottsay** [comment]
Fresh CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10842/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6215/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8833/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10735/)
